code/modules/research/machinery/_production.dm

```diff
@@ -279,19 +281,32 @@
 			return FALSE
 		borg.cell.use(SILICON_LATHE_TAX)
 
-	//consume materials
-	materials.use_materials(design.materials, coefficient, print_quantity, "built", "[design.name]")
-	//produce item
-	busy = TRUE
+	materials.use_materials(needed_materials, action = "built", name = "[design.name]")
 	if(production_animation)
 		flick(production_animation, src)
+
 	var/time_coefficient = design.lathe_time_factor * efficiency_coeff
-	addtimer(CALLBACK(src, PROC_REF(reset_busy)), (30 * time_coefficient * print_quantity) ** 0.5)
-	addtimer(CALLBACK(src, PROC_REF(do_print), design.build_path, print_quantity), (32 * time_coefficient * print_quantity) ** 0.8)
-	update_static_data_for_all_viewers()
+	var/build_time_per = (design.construction_time * time_coefficient) ** 0.8
+	start_making(design, print_quantity, user, build_time_per, materials_per_item)
 
 	return TRUE
 
+/obj/machinery/rnd/production/proc/start_making(datum/design/design, build_count, mob/user, build_time_per_item, list/materials_per_item)
```

Comment:
Autodoc with params. Maybe make it PRIVATE(TRUE) if not used anywhere but that's your call